src/url-handling.js

```diff
@@ -226,22 +266,80 @@ export function stateFromLocation(location: Location): UrlState {
       rangeFilters: query.range ? parseRangeFilters(query.range) : [],
       selectedThread: selectedThread,
       callTreeSearchString: query.search || '',
-      threadOrder: query.threadOrder
-        ? query.threadOrder.split('-').map(index => Number(index))
-        : [],
-      hiddenThreads: query.hiddenThreads
-        ? query.hiddenThreads.split('-').map(index => Number(index))
+      globalTrackOrder: query.globalTrackOrder
+        ? query.globalTrackOrder.split('-').map(index => Number(index))
         : [],
+      hiddenGlobalTracks: query.hiddenGlobalTracks
+        ? new Set(
+            query.hiddenGlobalTracks.split('-').map(index => Number(index))
+          )
+        : new Set(),
+      hiddenLocalTracksByPid: query.hiddenLocalTracksByPid
+        ? parseHiddenTracks(query.hiddenLocalTracksByPid)
+        : new Map(),
+      localTrackOrderByPid: query.localTrackOrderByPid
+        ? parseTrackOrder(query.localTrackOrderByPid)
+        : new Map(),
       markersSearchString: query.markerSearch || '',
       transforms,
     },
   };
 }
 
-type ProcessedLocation = { pathname: string, hash: string, query: Object };
+/**
+ * Hidden tracks must have the track indexes plus the associated PID.
+ *
+ * Syntax: Pid-TrackIndex-TrackIndex~Pid-TrackIndex
+ * Example: 124553-0-3~124554-1
+ */
+function parseHiddenTracks(rawText: string): Map<Pid, Set<TrackIndex>> {
+  const hiddenLocalTracksByPid = new Map();
+
+  for (const stringPart of rawText.split('~')) {
+    const [pidString, ...indexStrings] = stringPart.split('-');
+    if (indexStrings.length === 0) {
+      continue;
+    }
+    const pid = Number(pidString);
+    const indexes = indexStrings.map(string => Number(string));
+    if (!isNaN(pid) && indexes.every(n => !isNaN(n))) {
+      hiddenLocalTracksByPid.set(pid, new Set(indexes));
+    }
+  }
+  return hiddenLocalTracksByPid;
+}
+
+function parseTrackOrder(rawText: string): Map<Pid, TrackIndex[]> {
+  const localTrackOrderByPid = new Map();
+
+  for (const stringPart of rawText.split('~')) {
+    const [pidString, ...indexStrings] = stringPart.split('-');
+    if (indexStrings.length === 1) {
+      continue;
+    }
+    const pid = Number(pidString);
+    const indexes = indexStrings.map(string => Number(string));
+    if (!isNaN(pid) && indexes.every(n => !isNaN(n))) {
+      localTrackOrderByPid.set(pid, indexes);
+    }
+  }
+
+  return localTrackOrderByPid;
+}
```

Comment:
These 2 functions a very similar, wondering if this couldn't be just one ?

Reply:
I think it would be more complicated to conflate them, even though they do share similar checks. I think duplication is the simpler implementation solution here.